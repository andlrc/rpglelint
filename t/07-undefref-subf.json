[
   {
      "column":1,
      "file":"t/07-undefref-subf.rpgle",
      "line":13,
      "message":"'is_global_err' undeclared",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/07-undefref-subf.rpgle",
      "line":16,
      "message":"'is_subf_err' undeclared",
      "severity":"warning"
   },
   {
      "column":13,
      "file":"t/07-undefref-subf.rpgle",
      "line":19,
      "message":"'is_subf_err' undeclared",
      "severity":"warning"
   },
   {
      "column":1,
      "file":"t/07-undefref-subf.rpgle",
      "line":21,
      "message":"'my_ds_err' undeclared",
      "severity":"warning"
   },
   {
      "column":11,
      "file":"t/07-undefref-subf.rpgle",
      "line":21,
      "message":"'is_global' undeclared",
      "severity":"warning"
   },
   {
      "column":1,
      "file":"t/07-undefref-subf.rpgle",
      "line":22,
      "message":"'my_ds_err' undeclared",
      "severity":"warning"
   },
   {
      "column":11,
      "file":"t/07-undefref-subf.rpgle",
      "line":22,
      "message":"'is_global_arr' undeclared",
      "severity":"warning"
   }
]
