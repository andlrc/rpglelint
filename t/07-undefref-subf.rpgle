// rpglelint: -Wundefined-reference
dcl-ds my_ds;
  is_global int(10);
end-ds;

dcl-ds my_ds2 qualified;
  is_subf int(10);
end-ds;

dcl-ds my_ds2_like likeds(my_ds2);

is_global = 1;
is_global_err = 1;

my_ds2.is_subf = 1;
my_ds2.is_subf_err = 1;

my_ds2_like.is_subf = 1;
my_ds2_like.is_subf_err = 1;

my_ds_err.is_global = 1;
my_ds_err.is_global_arr = 1;
