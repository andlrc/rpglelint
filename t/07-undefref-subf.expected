t/07-undefref-subf.rpgle:13:1: warning: 'is_global_err' undeclared [-Wundefref]
 is_global_err = 1;
 ^
t/07-undefref-subf.rpgle:16:8: warning: 'is_subf_err' undeclared [-Wundefref]
 my_ds2.is_subf_err = 1;
        ^
t/07-undefref-subf.rpgle:19:13: warning: 'is_subf_err' undeclared [-Wundefref]
 my_ds2_like.is_subf_err = 1;
             ^
t/07-undefref-subf.rpgle:21:1: warning: 'my_ds_err' undeclared [-Wundefref]
 my_ds_err.is_global = 1;
 ^
t/07-undefref-subf.rpgle:22:1: warning: 'my_ds_err' undeclared [-Wundefref]
 my_ds_err.is_global_arr = 1;
 ^
